### ISSUE TITLE
aws-partitions requires Ruby version >= 2.3

### ISSUE THREAD

**Issue by v4de** [widdix/aws-s3-virusscan#89]
This issue started at 1am this morning for me when a message was placed in the scan queue and an instance was created to begin to process the queue. That message is still there and from the cloud init output log i see that it failed to initialize the instance:
`Error occurred during build: Command m_gem_install_aws_sdk_sqs failed`

I see this related to m_gem_install in the cfn:
 ```           
 m_gem_install_aws_sdk_sqs: # workaround for https://github.com/aws/aws-sdk-ruby/pull/2573#issuecomment-913984611
   command: 'gem install aws-sdk-sqs -v 1.23.1'
   test: '! gem list aws-sdk-sqs -i -v 1.23.1'
 ```

I used ssm to log into the instance and tried to install aws-sdk-sqs and get this error:
```
[ssm-user@ip-10-30-32-20 ~]$ gem install aws-sdk-sqs -v 1.23.1
Fetching: aws-sdk-sqs-1.23.1.gem (100%)
Successfully installed aws-sdk-sqs-1.23.1
Fetching: aws-partitions-1.522.0.gem (100%)
ERROR: Error installing aws-sdk-sqs:
aws-partitions requires Ruby version >= 2.3.
```

Do we need to upgrade the Ruby version?

**Comment by michaelwittig** [comment]
Duplicate of #88 #87
Please update to the latest version.

**Comment by v4de** [comment]
I have been using the latest version of the template since the dependency issue in September. It has been working fine until this morning. Just tried using the latest AMI's and nothing changed. Will try updating ruby and trying the same install on the instance.

**Comment by v4de** [comment]
@michaelwittig Resolved by upgrading to latest ruby 2.6 via amazon extras.
```
            a_enable_ruby2.6:
              command: amazon-linux-extras enable ruby2.6=stable && yum -y clean metadata
              test: '! grep -Fxq "[amzn2extra-ruby2.6]" /etc/yum.repos.d/amzn2-extras.repo'
```
adding rubygem json since it is not included with ruby 2.6
```
        config:
          packages:
            yum:
              clamd: []
              clamav: []
              clamav-update: []
              rubygem-json: []
```
and installing daemons and secure random via command
```
            gem_install_daemons:
              command: gem install daemons -v 1.3.1
              test: ! gem list daemons -i -v 1.3.1
            gem_install_rubysl-securerandom:
              command: gem install rubysl-securerandom -v 2.0.0
              test: ! gem list rubysl-securerandom -i -v 2.0.0
```
also removed the --ignore-dependencies flag from s3 sdk
```
            p_gem_install_aws_sdk_s3: # workaround for https://github.com/aws/aws-sdk-ruby/pull/2573#issuecomment-913984611
              command: gem install aws-sdk-s3 -v 1.54.0
              test: ! gem list aws-sdk-s3 -i -v 1.54.0
```
(also removed unnecessary and inconsistent single quotes)

**Comment by michaelwittig** [comment]
I was running into issues with `cfn-init` when updating ruby. I pushed a fix s well.
